src/ui/component/skeleton/skeleton.tsx

```diff
@@ -0,0 +1,35 @@
+import type { CSSProperties, FC } from 'react'
+import classNames from 'classnames'
+import './skeleton.scss'
+
+type SkeletonProps = {
+  width?: number | string
+  height?: number | string
+  variant?: 'text' | 'rectangular' | 'rounded' | 'circular'
+}
+
+export const Skeleton: FC<SkeletonProps> = ({ width, height, variant = 'text' }) => {
+  const mergedStyle: CSSProperties = {
+    width: width ? (typeof width === 'number' ? `${width}px` : width) : '100%',
+    ...(height ? { height: typeof height === 'number' ? `${height}px` : height } : {})
+  }
+
+  const isRounded = variant === 'rounded'
+  if (isRounded && !height) {
+    console.error(
+      "A height must be provided when using the 'rounded' variant of Skeleton component."
+    )
+  }
+  if (isRounded && height) {
+    const borderRadius =
+      typeof height === 'number' ? `${Math.round(height * 0.5)}px` : `calc(${height} * 0.5)`
+    mergedStyle.borderRadius = borderRadius
+  }
```

Comment:
I'm not sure you need all these:

```
const isRounded = variant === 'rounded'
  if (isRounded && !height) {
    console.error(
      "A height must be provided when using the 'rounded' variant of Skeleton component."
    )
  }
  if (isRounded && height) {
    const borderRadius =
      typeof height === 'number' ? `${Math.round(height * 0.5)}px` : `calc(${height} * 0.5)`
    mergedStyle.borderRadius = borderRadius
  }
```
  
 I tested without and it works well.
if the height is not specified, it takes the value `auto`, knowing that you have also defined a content with a value `" "` which allows you to have a content to display.
The border radius is defined at 10 px and can stay like that without calculating different border radius.
What do you think?

Reply:
@lolottetheclash default Skeleton variant is `text`, with an height of 1.2em and a 4px radius.
If we set a wrong variant name like foo, it fallbacks on a 10px radius.

If we remove the computed radius and not set a variant by default, we will end with a 10px radius by default. Is it working for most of use cases? Probably.
If you agree we can remove the code you mentioned.

Reply:
@0xLodz No pb for me to keep the default variant and also the 10px radius, also I don't think that we should have a fallback because the variants possibilities are defined and typed.
I think you can remove the code I copied previously.